analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ConditionEvaluatesToConstant.CSharp8.cs

```diff
@@ -267,38 +267,6 @@ public void EmptyCollectionCheck2()
         }
     }
 
-    public class PassingArgumentsByReference
-    {
-        private static object _gate = new object();
-
-        public void InitWithLock<T>(ref T field)
-            where T : new()
-        {
-            if (field == null)
-            {
-                lock (_gate)
-                {
-                    if (field == null) // Noncompliant, FP - in multithreading context it makes sense to check for null twice
-                    {
-                        field = new T();
-                    }
-                }
-            }
-        }
-
-        public void Init<T>(ref T field)
```

Comment:
sorry for changing my mind here, but I realize that this example documents one valid Noncompliant and one FP...

- the FP is where there's a `lock` and it seems we talk about a multi-threaded env
- the TP is where there's no `lock`, so no specific suggestion the code should be used in multi-threaded env

maybe we should just drop the PR?

Reply:
I moved the old FP example as we agreed on Slack